src/components/AppPageContainer.tsx

```diff
@@ -3,21 +3,22 @@ import Footer, { footerBreakpoint, footerMarginTop } from 'components/Footer';
 import { PropsWithChildren, useEffect, useState } from 'react';
 
 import UserContext from "../UserContext";
-import trpc from "../trpc";
 import User from '../shared/User';
 import NavBars, { sidebarBreakpoint, sidebarContentMarginTop, topbarHeight } from 'components/Navbars';
 import PageLoader from 'components/PageLoader';
+import { useSession } from 'next-auth/react';
 
 export default function AppPageContainer({ children, wide, ...rest }: {
   wide: boolean
 } & PropsWithChildren) {
   const [user, setUser] = useState<User | null>(null);
+  const { data: session } = useSession();
```

Comment:
don't call useSession() here. pass in the session object from _app.tsx:SwitchBoard()

Reply:
The session has been passed down properly with the provider already, so a getSession is used instead. 
**Mark** that getSession is async and to be used when user data is required

Reply:
I'm not sure I understand your comment.

Reply:
The session was already properly passed by 
```javascript 
<SessionProvider session={session} />
``` 
So I meant that I do not need to address the `pass in the session object from _app.tsxLSwitchBoard()` mentioned in the previous comment, and you should just call `getSession()` API whenever the client side needs to use the user session data, I will make a note in this PR comment

Reply:
1. when useSession() is available, we should use it instead of getSession(). The former is the proper "React way" to do it.

2. The reason I am requesting passing in session from SwitchBoard() is to make the session object being not null as an invariant to AppPageContainer. Invariants are important for code maintainability.